### ISSUE TITLE
much larger size between 3.1.30 and 3.1.2.1

### ISSUE THREAD

**Issue by bgutman** [just-containers/s6-overlay#507]
From READ.ME under performance it says it should be only 6.5MB.  The base Busybox image is 4.8MB but after building docker images with 3.13.0 compared to 3.1.2.1  The docker images are different in size, 495MB compared to 11.7MB

Version 3.1.3.0 the resulting docker image has 495MB size

FROM busybox
ADD https://github.com/just-containers/s6-overlay/releases/download/v3.1.3.0/s6-overlay-noarch.tar.xz /tmp
RUN tar -C / -Jxpf /tmp/s6-overlay-noarch.tar.xz
ADD https://github.com/just-containers/s6-overlay/releases/download/v3.1.3.0/s6-overlay-x86_64.tar.xz /tmp
RUN tar -C / -Jxpf /tmp/s6-overlay-x86_64.tar.xz
ENTRYPOINT ["/init"]

Version 3.1.2.1 the resulting docker image has 11.7MB size

FROM busybox
ADD https://github.com/just-containers/s6-overlay/releases/download/v3.1.2.1/s6-overlay-noarch.tar.xz /tmp
RUN tar -C / -Jxpf /tmp/s6-overlay-noarch.tar.xz
ADD https://github.com/just-containers/s6-overlay/releases/download/v3.1.2.1/s6-overlay-x86_64.tar.xz /tmp
RUN tar -C / -Jxpf /tmp/s6-overlay-x86_64.tar.xz
ENTRYPOINT ["/init"]

**Comment by nekopsykose** [comment]
seems to be caused by every `command` being 2MB, of which:  

```
bloaty ./execline-2.9.1.0/command/case
    FILE SIZE        VM SIZE    
 --------------  -------------- 
  96.7%  1.94Mi   0.0%       0    [Unmapped]
   2.3%  47.8Ki  69.7%  47.8Ki    .text
   0.9%  17.6Ki  25.6%  17.6Ki    .rodata
   0.0%       0   3.8%  2.61Ki    .bss
   0.0%     460   0.7%     460    [LOAD #0 [RX]]
   0.0%     184   0.0%       0    .shstrtab
   0.0%     128   0.1%      64    .data
   0.0%     128   0.0%       0    [ELF Headers]
   0.0%     112   0.1%      48    .note.gnu.property
   0.0%     100   0.1%      36    .note.gnu.build-id
   0.0%      72   0.0%       8    .fini_array
   0.0%      72   0.0%       8    .init_array
   0.0%      68   0.0%       4    .eh_frame
   0.0%      67   0.0%       3    .fini
   0.0%      67   0.0%       3    .init
 100.0%  2.00Mi 100.0%  68.6Ki    TOTAL
```

so basically they are all padded to 2MB for some reason. of course this still compresses well, so it's not visible in the tarball. last i saw something like this it was a binutils linker bug for e.g. aarch64 padding everything, so maybe it's something like that.

running `strip` with binutils 2.39 myself fixes it. did you change your toolchains recently @skarnet ?

**Comment by skarnet** [comment]
Yeah, I changed the toolchain to gcc-12.2.0, with binutils-2.39. I only checked the tarballs, not the binaries themselves... and had not noticed that the toolchain was building binaries so large. My bad! Going to make a new toolchain with binutils-2.40 and see if it fixes it (and if it doesn't I'll revert to the previous toolchain), then push a new release.

Thanks for the report! That's an embarrassing issue 😅
